mcs/class/corlib/ReferenceSources/CompareInfo.cs

```diff
@@ -33,34 +33,69 @@
 
 namespace System.Globalization
 {
+	interface ISimpleCollator
+	{
+		SortKey GetSortKey (string source, CompareOptions options);
+
+		int Compare (string s1, string s2);
+
+		int Compare (string s1, int idx1, int len1, string s2, int idx2, int len2, CompareOptions options);
+
+		bool IsPrefix (string src, string target, CompareOptions opt);
+
+		bool IsSuffix (string src, string target, CompareOptions opt);
+
+		int IndexOf (string s, string target, int start, int length, CompareOptions opt);
+
+		int IndexOf (string s, char target, int start, int length, CompareOptions opt);
+
+		int LastIndexOf (string s, string target, CompareOptions opt);
+
+		int LastIndexOf (string s, string target, int start, int length, CompareOptions opt);
+
+		int LastIndexOf (string s, char target, CompareOptions opt);
+
+		int LastIndexOf (string s, char target, int start, int length, CompareOptions opt);
+	}
+
 	partial class CompareInfo
 	{
 		[NonSerialized]
-		SimpleCollator collator;
+		ISimpleCollator collator;
 
 		// Maps culture IDs to SimpleCollator objects
-		static Dictionary<string, SimpleCollator> collators;
+		static Dictionary<string, ISimpleCollator> collators;
 		static bool managedCollation;
 		static bool managedCollationChecked;
 
 		static bool UseManagedCollation {
 			get {
+#pragma warning disable 162
+				if (!GlobalizationSupport.SupportsManagedCollation)
+					return false;
 				if (!managedCollationChecked) {
 					managedCollation = Environment.internalGetEnvironmentVariable ("MONO_DISABLE_MANAGED_COLLATION") != "yes" && MSCompatUnicodeTable.IsReady;
 					managedCollationChecked = true;
 				}
 
 				return managedCollation;
+#pragma warning restore 162
 			}
 		}
 
-		SimpleCollator GetCollator ()
+		ISimpleCollator GetCollator ()
 		{
+#pragma warning disable 162
+			// This is `const bool SupportsManagedCollation = false` on WASM.
+			if (!GlobalizationSupport.SupportsManagedCollation)
```

Comment:
Should not be needed with the change above

Reply:
Unreachable code still needs to compile, so we can't remove this method or the interface without also `#if`-def'ing everything that calls it.

However, at least for the time being I would like to keep the compiled output as closely to what my linker module would yield as output.